src/utils/getElementId.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * SPDX-FileCopyrightText: 2018 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import md5 from 'md5'
+
+// unique hash for this app and library version (make sure that there is no clash with different apps)
+const instanceId = md5(`${appName}@${SCOPE_VERSION}`).slice(0, 5)
+let id = 0
+
+/**
+ * Get the next element id for use with HTML / CSS.
+ * We can not be sure that the app has a prefix setup so we prefix ourself
+ */
+export function getElementId() {
+	return `nc${instanceId}-${id++}`
+}
```

Comment:
Hmm, ok, in this case it makes sense not to use `Math.random()` for a stable result.

But still I'm not fun of `md5` hash.

Can we sanitize the app name instead of hashing it? No dependency and more readable in the layout.

Reply:
`window.crypto` `getRandomValues() or `randomUUID()` could be applied here instead, maybe?

Reply:
The idea of `md5` here is to hash a constant value (and have a stable ID sequence). For a random value we can simply use `Math.random()`.

Crypto API has methods to generate hashes as well, but they are async.

Reply:
I was thinking about something like
```js
appName
  .replace(/\s+/g, '-')
  .replace(/[^\w-]/g, '')
```

Reply:
Still a problem if there is a library using a different nc-vue version and also providing components like nc-upload or nc-password confirmation.

I think I found a solution by using a global id?

Reply:
> I think I found a solution by using a global id?

Ah, so simple and so elegant